core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -96,6 +96,30 @@ class DAGScheduler(
   // Stages that must be resubmitted due to fetch failures
   private[scheduler] val failedStages = new HashSet[Stage]
 
+  // The maximum number of times to retry a stage before aborting
+  val maxStageFailures = 5
```

Comment:
can you make this a conf?  there is already `spark.task.maxFailures`, so how about `spark.stage.maxFailures`?  Also it should get added to the docs